src/common/shared_config.h

```diff
@@ -7,32 +7,23 @@
 // bootloader version
 #define BOOTLOADER_VERSION_ADDRESS 0x0801FFFA
 
-// EEPROM firmware update flag
-#define FW_UPDATE_ENABLE  0xAA
-#define FW_UPDATE_DISABLE 0x00
-
-// pin PA13 state
-#define APPENDIX_FLAG_MASK 0x01
+typedef enum {
+    BT_APPENDIX_EXISTS = 0, /**< Appendix not broken, signature authentication needed */
+    BT_APPENDIX_BROKEN /**< Appendix broken, no signature authentication needed */
+} BT_APPENDIX_STATUS;
 
 #pragma pack(push)
 #pragma pack(1)
-typedef struct {
-    uint8_t fw_update_flag;
-    uint8_t model_specific_flags; // ~ "reserved1" originally
-    uint8_t reserved2;
-    uint8_t reserved3;
-    uint8_t reserved4;
-    uint8_t reserved5;
-    uint8_t reserved6;
-    uint8_t reserved7;
-    uint8_t reserved8;
-    uint8_t reserved9;
-    uint8_t reserved10;
-    uint8_t reserved11;
-    uint8_t reserved12;
-    uint8_t reserved13;
-    uint8_t reserved14;
-    uint8_t reserved15;
+
+typedef enum {
+    FW_UPDATE_ENABLE = 0xAA,
+    FW_UPDATE_DISABLE = 0xFF,
+} fw_update_flag;
+
+typedef struct  {
+    fw_update_flag       fw_update_flag;
+    bool                 appendix_broken : 1;
```

Comment:
Using bool this way is very unusual. I'm not saying the gcc won't accept this, but using the good old uint8_t makes things more clear. Bool may actually be represented as 4B or even 8B variable depending on target platform.

Reply:
good point!